components/CurrentLocation.js

```diff
@@ -98,17 +105,68 @@ function CurrentLocation () {
         return () => clearInterval(intervalId);
     });
 
-    currentFloor = (altitude - baseAltitude) / floorHeight;
+    //Compute the floor level
+    // currentFloor = (altitude - baseAltitude) / floorHeight;
+    // roundedCurrentFloor = "Floor level: " + (Math.round(currentFloor * 100) / 100).toFixed(0);
 
-    if (currentFloor > 0) {
-        roundedCurrentFloor = "This is the " + (Math.round(currentFloor * 100) / 100).toFixed(0) + "rd floor";
-    }
 
     if (currentBuilding === "") {
```

Comment:
Instead of a big if else statement for something redundant, you can use conditional rendering. The reason why is because the same components are rendered, the only difference is lines 119 & 134/135. So, I would recommend this:

```
return (
<View style={styles.view}>
 <Icon style={styles.icon} name='location' size="30"  color="#2A2E43" onPress={() => setModalVisibility(true)}/>
     <Modal isVisible={modalVisibility}>
        {currentBuilding === "" &&
           <View style={styles.modal}>
             <Text style={styles.modalText}>Please try again in a Concordia building</Text>
             <Button style={styles.modalButton} title="Close" onPress={() => setModalVisibility(false)}/>
           </View>
        }
        {currentBuilding !== "" &&
            <View style={styles.modal}>
                <Text style={styles.modalText}>{currentBuilding}</Text>
                <Text style={styles.modalText}>{/*roundedCurrentFloor*/}2nd floor</Text>
                <Button style={styles.modalButton} title="Close" onPress={() => setModalVisibility(false)}/>
            </View>
        }
    </Modal>
</View>
);

Reply:
Very great catch... But we still have redundant code with your solution. @mohanadarafe  Your solution only remove the need to have two returns, but as you pointed out, it looks like it's the same component. Also, his _touchableOpacity_ is not taken into consideration in your solution. Still, on top of what you suggested, I suggested creating a component corresponding to 
`
        
        <TouchableOpacity>
        <Icon style={styles.icon} name='location' size="30"  color="#2A2E43" onPress={() => setModalVisibility(true)}/>
        </TouchableOpacity>
        <Modal isVisible={modalVisibility}>
            <View style={styles.modal}>
                <Text style={styles.modalText}>{currentBuilding}</Text>
                <Text style={styles.modalText}>{/*roundedCurrentFloor*/}2nd floor</Text>
                <Button style={styles.modalButton} title="Close" onPress={() => setModalVisibility(false)}/>
            </View>
        </Modal>
        </View>
This way, you can simply pass the props you need into this component so it renders what you need. The component could have the touchableopacity set to disabled or not visible and becomes active when your component receives the props to activated it. Same for the extra `<Text style={styles.modalText}>{/*roundedCurrentFloor*/}2nd floor</Text>` . You can Slack me if you don't quite understand my suggestion.

Reply:
Alain i understand your solution but i feel like mohanad's solution is better for us since it is easier to understand. Not only for us the dev team, but also for the people who are taking care of testing. I want the code to be as easy to read as possible and i feel like mohanad's version is better for that.

Reply:
I didn't read your reply. Sure. But I suggested something different because I thought it wasn't that easy to read...  Conditional rendering is PERFECT when it's not multiple lines. And adding components is not a bad thing.